tools/check/check-timeout.go

```diff
@@ -0,0 +1,165 @@
+		"testEvaluatorSuite.TestSleepVectorized",
+		"testSuite.TestFailNewSession",
+		"testSuite.TestFailNewSession",
+		"testPlanSerialSuite.TestPartitionTable",
+		"testPlanSerialSuite.TestPartitionWithVariedDatasources",
+		"testPrivilegeSuite.TestSecurityEnhancedModeInfoschema",
+		"HTTPHandlerTestSuite.TestZipInfoForSQL",
+		"HTTPHandlerTestSuite.TestBinlogRecover",
+		"ConnTestSuite.TestConnExecutionTimeout",
+		"ConnTestSuite.TestTiFlashFallback",
+		"tidbTestTopSQLSuite.TestTopSQLCPUProfile",
+		"testPessimisticSuite.TestAmendForIndexChange",
+		"testPessimisticSuite.TestGenerateColPointGet",
+		"testPessimisticSuite.TestInnodbLockWaitTimeout",
+		"testPessimisticSuite.TestPessimisticLockNonExistsKey",
+		"testPessimisticSuite.TestSelectForUpdateNoWait",
+		"testSessionSerialSuite.TestProcessInfoIssue22068",
+		"testKVSuite.TestRetryOpenStore",
+		"testBatchCopSuite.TestStoreErr",
+		"testBatchCopSuite.TestStoreSwitchPeer",
+		"testSequenceSuite.TestSequenceFunction",
+		"testSuiteP2.TestUnion",
+		"testVectorizeSuite1.TestVectorizedBuiltinTimeFuncGenerated",
+		"testSuiteJoin3.TestVectorizedShuffleMergeJoin",
+		"testSuite1.TestClusterIndexAnalyze",
+		"testSuiteJoin3.TestVectorizedMergeJoin",
+		"testFastAnalyze.TestAnalyzeFastSample",
+		"testBypassSuite.TestLatch",
+	}
+	for _, v := range tmp {
+		allowList[v] = struct{}{}
+	}
+}
+
+func inAllowList(input string) bool {
+	// The input looks like this.
+	// PASS: sysvar_test.go:131: testSysVarSuite.TestIntValidation
+	offset := strings.LastIndexByte(input, ' ')
+	if offset < 0 {
+		return false
+	}
+	testName := input[offset+1:]
+	_, ok := allowList[testName]
+	return ok
+}
+
+func main() {
+	lines := make([]string, 0, 100)
+	// The input looks like that:
+	// PASS: sysvar_test.go:131: testSysVarSuite.TestIntValidation	0.000s
+	// PASS: sysvar_test.go:454: testSysVarSuite.TestIsNoop	0.000s
+	// PASS: sysvar_test.go:302: testSysVarSuite.TestMaxExecutionTime	0.000s
+	// PASS: sysvar_test.go:429: testSysVarSuite.TestReadOnlyNoop	0.000s
+	// PASS: sysvar_test.go:97: testSysVarSuite.TestRegistrationOfNewSysVar	0.000s
+	// PASS: sysvar_test.go:269: testSysVarSuite.TestSQLModeVar	0.000s
+	// PASS: sysvar_test.go:254: testSysVarSuite.TestSQLSelectLimit	0.000s
+	// PASS: sysvar_test.go:224: testSysVarSuite.TestScope	0.000s
+	// PASS: sysvar_test.go:439: testSysVarSuite.TestSkipInit	0.000s
+	//
+	// They are generated by "grep '^:PASS' gotest.log", and gotest.log is generated by
+	// 'make gotest' from the Makefile which runs 'go test -v ./...' alike
+	scanner := bufio.NewScanner(os.Stdin)
+	for scanner.Scan() {
+		line := scanner.Text()
+		offset := strings.LastIndexByte(line, '\t')
+		if offset < 0 {
+			fmt.Println("get duration string error:", line)
+			os.Exit(-1)
+		}
+		durStr := line[offset+1:]
+		dur, err := time.ParseDuration(durStr)
+		if err != nil {
+			fmt.Println("parse duration string error:", line, err)
+			os.Exit(-2)
+		}
+
+		if dur > 4*time.Second {
```

Comment:
pr title mentioned it as 4 seconds

Reply:
Yes, the problem is, in my own computer, some test can finish in 2.Xs, while in the CI environment, it takes more than 3s ...
So to make the CI more stable, I change here to 4s ... (If some one find his test case can finish in his own computer, but fail in CI, he may blame the environment instead of improving his code...)